nmdc_schema/migrators/migration_reporter.py

```diff
@@ -0,0 +1,261 @@
+            self.logger.info(row)
+        
+        self.logger.info("")
+    
+    def _generate_unmapped_values_table(self) -> None:
+        """Generate the unmapped values table."""
+        self.logger.info("UNMAPPED VALUES")
+        self.logger.info("-" * 40)
+        
+        if not self.unmapped_values:
+            self.logger.info("No unmapped values found.")
+            self.logger.info("")
+            return
+        
+        # Table header
+        self.logger.info(f"{'Class.Slot':<35} {'Unmapped Value':<20} {'Total Count':<12}")
+        self.logger.info("-" * 67)
+        
+        # Sort by class.slot, then by count (descending)
+        for class_slot in sorted(self.unmapped_values.keys()):
+            values = self.unmapped_values[class_slot]
+            sorted_values = sorted(values.items(), key=lambda x: -x[1])
+            
+            for value, count in sorted_values:
+                self.logger.info(f"{class_slot:<35} {value:<20} {count:<12}")
+        
+        self.logger.info("")
+    
+    def _generate_missing_values_table(self) -> None:
+        """Generate the missing values table."""
+        self.logger.info("MISSING VALUES")
+        self.logger.info("-" * 40)
+        
+        if not self.missing_values:
+            self.logger.info("No missing values found.")
+            self.logger.info("")
+            return
+        
+        # Table header
+        self.logger.info(f"{'Class.Slot':<35} {'Total Missing':<12}")
+        self.logger.info("-" * 47)
+        
+        # Sort by class.slot
+        for class_slot in sorted(self.missing_values.keys()):
+            count = self.missing_values[class_slot]
+            self.logger.info(f"{class_slot:<35} {count:<12}")
+        
+        self.logger.info("")
+    
+
+
+# Generic schema path utilities for migration reporting
+def parse_schema_path(path: str) -> List[str]:
+    """
+    Parses a document path into schema-relevant components, filtering out array indices.
+
+    Args:
+        path: Path like "substances_used[0].volume" or "extraction.input_mass"
+
+    Returns:
+        List of schema slot names: ["substances_used", "volume"] or ["extraction", "input_mass"]
+    """
+    if not path:
+        return []
+
+    parts = []
+    for part in path.split('.'):
+        if '[' in part:
+            # Extract slot name, ignore array index
+            slot_name = part.split('[')[0]
+            if slot_name:  # Only add non-empty slot names
+                parts.append(slot_name)
+        else:
+            parts.append(part)
+
+    return parts
+
+
+def get_clean_schema_path(path: str) -> str:
+    """
+    Converts a document path with array indices to a clean schema path for reporting.
+    
+    Args:
+        path: Path like "substances_used[0].volume" or "extraction.input_mass"
+        
+    Returns:
+        Clean schema path: "substances_used.volume" or "extraction.input_mass"
+    """
```

Comment:
I suggest adding some doctests to this helper function (I confirmed these pass).

```suggestion

    >>> get_clean_schema_path("extraction.input_mass")  # no array
    'extraction.input_mass'
    >>> get_clean_schema_path("substances_used[0].volume")  # 1-D array
    'substances_used.volume'
    """
```